internal/kafka/internal/services/cluster_placement_strategy.go

```diff
@@ -68,7 +67,7 @@ func (f *FirstSchedulableWithinLimit) FindCluster(kafka *dbapi.KafkaRequest) (*a
 
 	kafkaInstanceSize, e := f.KafkaConfig.GetKafkaInstanceSize(kafka.InstanceType, kafka.SizeId)
 	if e != nil {
-		return nil, errors.NewWithCause(errors.ErrorInstancePlanNotSupported, e, "failed to find cluster with criteria '%v'", criteria)
+		return nil, e
```

Comment:
Why this change?

Reply:
These changes were due to the change of returned error type in the cluster placement strategy. All of the internal calls now returns an `error` type so we no longer need to create a `ServiceError` out of them. 

We can wrap these errors with the original message, but note that most of these will be database failures.

Reply:
A follow up on @ziccardi question.

> All of the internal calls now returns an error type so we no longer need to create a ServiceError out of them.

the `*errors.ServiceError` implements the `error` interface, I think it'd have been fine to only leave the returned service error.

 > but note that most of these will be database failures
 
 Just to confirm that the database failures are not plainly exposed to the end users, correct? to avoid leaking out db credentials or something ;-)

Reply:
> the *errors.ServiceError implements the error interface, I think it'd have been fine to only leave the returned service error and only change the method signature.

I assumed that we shouldn't mix that two even if we can, maybe I misunderstood that though :) 

> Just to confirm that the database failures are not plainly exposed to the end users, correct? to avoid leaking out db credentials or something ;-)

I think we were wrapping them originally in the Service Errors but only logged them, they're not exposed to users. I believe we return a general error message in the handlers.

Reply:
> I assumed that we shouldn't mix that two even if we can, maybe I misunderstood that though :)

I was thinking only the method signature may be impacted and we as well as return the service error which is just an implementation of the `error` interface, without any issue. I think we can safely return it and only change the error type from the method signature.

PS: no need to revert the changes here due to explanations you've given above.

Reply:
> I think we were wrapping them originally in the Service Errors but only logged them, they're not exposed to users. I believe we return a general error message in the handlers.

Thanks.

Reply:
> I assumed that we shouldn't mix that two even if we can, maybe I misunderstood that though :)

What is important is that the returned value is an `error`, which is just an interface.
The returned concrete object can be anything you want as long as it implements `error`.